core/src/main/java/org/apache/camel/kafkaconnector/utils/CamelKafkaConnectMain.java

```diff
@@ -227,7 +229,10 @@ public CamelKafkaConnectMain build(CamelContext camelContext) {
             camelMain.configure().setDumpRoutes(Boolean.TRUE.toString());
 
             Properties camelProperties = new Properties();
-            camelProperties.putAll(props);
+            Map<String, String> camelProps = props.entrySet().stream()
```

Comment:
The problem here is that, theoretically (since it is not explicitly tested in tests, would be possible to add properties not starting with `camel.` as a way to configure some other useful stuff like factories for some camel components i.e. as in https://github.com/apache/camel-kafka-connector/blob/main/connectors/camel-kafka-connector-fix-dependencies.properties#L25 what about having a specific property say `camel.kafkaconnector.filtered.props` as a comma separated list of sensitive properties that one may want to explicitly filter?